qa/pull-tester/rpc-tests.py

```diff
@@ -266,7 +268,25 @@ def main():
         subprocess.check_call((config["environment"]["SRCDIR"] + '/qa/rpc-tests/' + test_list[0]).split() + ['-h'])
         sys.exit(0)
 
-    run_tests(test_list, config["environment"]["SRCDIR"], config["environment"]["BUILDDIR"], config["environment"]["EXEEXT"], args.jobs, args.coverage, passon_args)
+
+    if (args.rpcgroup == -1) != (args.machines == -1):
+        print("ERROR: Please use both -m and -r options when using parallel rpc_groups.")
+        sys.exit(0)
+    if args.machines == 0:
+        print("ERROR: -m/--machines must be greater than 0 and -r/--rpcgroup must be less than -m/--machines")
+        sys.exit(0)
+    if args.rpcgroup > args.machines:
+        print("ERROR: -r/--rpcgroup must be less than -m/--machines")
+        sys.exit(0)
+    if args.rpcgroup != -1 and args.machines != -1 and args.machines > args.rpcgroup:
```

Comment:
These error messages no longer match the actual behaviour. Currently `args.machines == args.rpcgroup` is equivalent to not specifying either argument (i.e. run all tests), but the errors assume the prior behaviour where `args.rpcgroup >= args.machines` is treated as an error. What is the intended behaviour?

Reply:
I added the `and -r/--rpcgroup must be less than -m/--machines")` just to help users provide a valid 2 param, but it makes the code a bit wonky to read. Should I just remove the `and -r/--rpcgroup must be less than -m/--machines")` ?

Reply:
I removed it since it makes it look like copy paste or bug issue since code does not match